Dockerfile

```diff
@@ -2,6 +2,18 @@ ARG ROS_DISTRO
 
 FROM ros:${ROS_DISTRO}-ros-base
 
+ARG DOCKER_USER=orise
+
+RUN useradd -s /bin/bash ${DOCKER_USER}
+
+# install gosu
+RUN set -eux; \
+    apt-get update; \
+    apt-get install -y gosu; \
+    rm -rf /var/lib/apt/lists/*; \
+    gosu nobody true
+
+# install common dev tools
```

Comment:
to be able to use root with user mode we need to modify the sudoer file like
```Dockerfile
RUN echo "${DOCKER_USER} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/${DOCKER_USER} && \
 chmod 0440 /etc/sudoers.d/${DOCKER_USER}
```

Reply:
using usermod would not also work?

Reply:
No, because it would need to set a password for the user, to be root without password you need the `NOPASSWD:AL` specification